utils/utils.py

```diff
@@ -40,8 +40,13 @@ def setup_ddp():
     distributed_data_parallelism = False
     world_size, world_rank = get_comm_size_and_rank()
 
-    master_addr = "127.0.0.1"
-    master_port = "8889"
+    ## source: https://www.olcf.ornl.gov/wp-content/uploads/2019/12/Scaling-DL-on-Summit.pdf
+    ## The following is Summit specific
+    import subprocess
+    get_master = "echo $(cat {} | sort | uniq | grep -v batch | grep -v login | head -1 )".format(os.environ['LSB_DJOB_HOSTFILE'])
```

Comment:
Can we add some subprocess call to check if we're on Summit before we do this? Then default to what we had before

Reply:
What will be the best way to check if the machine is Summit or not?

Reply:
@jychoi-hpc @streeve 

I was thinking to actually use "OMPI_COMM_WORLD_LOCAL_RANK" as suggested by Junqi. 
So I would do this:

local_device_rank = OMPI_COMM_WORLD_LOCAL_RANK
device_name = "cuda:" + str(local_device_rank)

Reply:
Right. I changed.